standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/dataconnector/jdbc/MySQLConnectorProvider.java

```diff
@@ -27,14 +27,27 @@
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
+import java.util.Properties;
 
 public class MySQLConnectorProvider extends AbstractJDBCConnectorProvider {
   private static Logger LOG = LoggerFactory.getLogger(MySQLConnectorProvider.class);
 
   private static final String DRIVER_CLASS = "com.mysql.jdbc.Driver";
 
+  public static final String JDBC_MYSQL_CONFIG_PREFIX = "hive.sql.mysql";
```

Comment:
Could we move these to parent `getConnectionProperties()`. For connection properties, all start with `hive.sql.connect.` for example, so users can configure the properties as they want.

Should we also take care of other database vendors?

Reply:
I have moved these three connection properties from `MySQLConnectorProvider` to the parent class `AbstractJDBCConnectorProvider`, so that others could reuse this properties.

Currentle, these three connection properties are only used by MySQL connector, because i am not sure other jdbc connectors have these properties.

Reply:
In order to be more generic, the connection properties can defined like this:
`hive.connector.key1=value1`
so when we `getConnectionProperties`, we filter all properties that start with `hive.connector.`,  retrieve the key from the `hive.connector.key1`, then put the `key1` and `value1` into the map.
So by this way, we can dynamically set the connection properties of different connectors without making special changes.

Reply:
Good suggestion. Please take another look. thx.